### PR TITLE
Product generator

### PR DESCRIPTION
**Pull request by mrijk**
Code generator, no impact on core functionality.

### CODE REVIEW

File: orchestrator/cli/generator/generator/migration.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import structlog
+
+from orchestrator.cli.generator.generator.helpers import get_variable
+
+logger = structlog.getLogger(__name__)
+
+
+def create_migration_file(config: dict) -> Optional[str]:
+    create_migration_command = f'PYTHONPATH=. DATABASE_URI=postgresql://nwa:nwa@localhost/nwa-workflows python main.py db revision --message "Add {config["name"]}" --head=data@head'
+    result = subprocess.check_output(create_migration_command, shell=True, text=True)  # noqa: S602
```
Comment (Mark90) [inline orchestrator/cli/generator/generator/migration.py:26]: Is it possible to import the typer/click command and invoke it programmatically?

File: orchestrator/cli/generator/generator/translations.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        workflows = {
+            f"create_{variable}": f"Create {name}",
+            f"modify_{variable}": f"Modify {name}",
+            f"validate_{variable}": f"Validate {name}",
+            f"terminate_{variable}": f"Terminate {name}",
+        }
+        translations["workflow"] = translations["workflow"] | workflows
+
+        path = ""
+        writer(path, translations["workflow"])
```
Comment (Mark90) [inline orchestrator/cli/generator/generator/translations.py:52]: Empty path?

File: orchestrator/cli/generator/generator/workflow.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+    generate_shared_workflow_files(environment, config, writer)
+    generate_create_workflow(environment, config, writer)
+    generate_modify_workflow(environment, config, writer)
+    generate_validate_workflow(environment, config, writer)
+    generate_terminate_workflow(environment, config, writer)
+
+    add_workflow_translations(config, writer)
+
+    # insert_lazy_workflow_instances(environment, config, writer)
```
Comment (Mark90) [inline orchestrator/cli/generator/generator/workflow.py:84]: Todo?

File: orchestrator/cli/generator/templates/macros.j2 [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{% macro optional(field) -%}
+{% if python_version == "3.9" -%}
+Optional[{{ field.type }}] = None
+{%- else -%}
+{{ field.type }} | None = {% if field.default is defined %} {{ field.default }}{% else %} None {% endif %}
+{%- endif -%}
```
Comment (Mark90) [inline orchestrator/cli/generator/templates/macros.j2:6]: The 3.9 branch does not have a default value

**Review by Mark90 — APPROVED** [review]
🚀

**Review by pboers1988 — APPROVED** [review]

**Pull request merged**